File: OpenDSS_files/MATLAB_Python_OpenDSS/generateOpenDSSlineScript.m
function openDSS_script = generateOpenDSSlineScript(line_data, phases)
    % Initialize an empty string to hold the full script
    openDSS_script = "";
    
    % Iterate over each line in the line_data
    for i = 1:size(line_data, 1)
        % Extract data for each line
        sending = line_data(i, 1);
        receiving = line_data(i, 2);
        R = line_data(i, 3);
        X = line_data(i, 4);
        
        % Format the OpenDSS command for this line
        line_command = sprintf('New Line.Line%d bus1=%d bus2=%d phases=%d R1=%f X1=%f\n', ...
                               i, sending, receiving, phases, R, X);
        
        % Append the line command to the full script, adding a new line after each entry
        openDSS_script = openDSS_script + line_command;
    end
    
    % Display the full OpenDSS script in the command window
    disp(openDSS_script)
    
end

File: OpenDSS_files/MATLAB_Python_OpenDSS/generateOpenDSSlineScript.py
# -*- coding: utf-8 -*-
"""
Created on Sun Nov  3 14:59:02 2024

@author: Morufdeen ATILOLA
"""

def generate_opendss_line_script(line_data, phases):
    # Initialize an empty string to hold the full script
    opendss_script = ""
    
    # Iterate over each row in the line_data
    for i, row in enumerate(line_data, start=1):
        sending, receiving, R, X = row  # Unpack the row
        
        # Format the OpenDSS command for this line
        line_command = f"New Line.Line{i} bus1={int(sending)} bus2={int(receiving)} phases={phases} R1={R:.4f} X1={X:.4f}\n"
        
        # Append the line command to the full script
        opendss_script += line_command
    
    return opendss_script